social-tw/social-tw-website
# [Frontend] Fix: modify randomNonce role

## Summary

This PR updates the randomNonce function to correctly generate nonces based on the user's action count. It ensures that the first three nonces (0, 1, 2) are used in order for the initial three actions, and subsequent actions receive a randomly chosen nonce within the same range.

## Linked Issue

fix #269 

## Details

- Updated `randomNonce` to consider the action count before deciding whether to return a sequential nonce or a random nonce.
- Imported `NUM_EPOCH_KEY_NONCE_PER_EPOCH` from config to use as the upper limit for nonce generation.

## Impacted Areas

- `packages/frontend/src/utils/randomNonce.ts`
- `packages/frontend/src/__tests__/utils/randomNonce.test.ts`

## Tests

- Unit tests have been updated in randomNonce.test.ts to cover the new logic.
- Tests ensure the function returns sequential nonces for the first three actions and random nonces thereafter.
- No specific environment or configuration needed apart from the standard project setup.

## Possible Impacts

- This change ensures more predictable and correct nonce generation for the first three user actions, impacting how the application handles these initial transactions.
- No backward compatibility issues are expected.

## Visual Materials

N/A

## Verification Steps

- Run the updated unit tests in randomNonce.test.ts to verify the correct function behavior.
- Integrate the function into the frontend and perform manual testing to observe nonce generation based on action counts.

## Todo

Break down the progress of the PR for everyone to see what else you intend to include in this PR.

-   [x] Update randomNonce function logic.
-   [x] Update and verify unit tests for randomNonce.

## Checklist
